File: main/contrib/Windows API Code Pack 1.1/source/Samples/TaskDialogDemo/VB/TaskDialogDemo/TestHarness.vb
'Copyright (c) Microsoft Corporation.  All rights reserved.


Imports Microsoft.VisualBasic
Imports System
Imports System.Windows.Forms
Imports Microsoft.WindowsAPICodePack
Imports Microsoft.WindowsAPICodePack.Dialogs

Namespace TaskDialogDemo
	Partial Public Class TestHarness
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub cmdShow_Click(ByVal sender As Object, ByVal e As EventArgs) Handles cmdShow.Click
			Dim td As New TaskDialog()

'			#Region "Button(s)"

			Dim button As TaskDialogStandardButtons = TaskDialogStandardButtons.None

			If chkOK.Checked Then
				button = button Or TaskDialogStandardButtons.Ok
			End If
			If chkCancel.Checked Then
				button = button Or TaskDialogStandardButtons.Cancel
			End If

			If chkYes.Checked Then
				button = button Or TaskDialogStandardButtons.Yes
			End If
			If chkNo.Checked Then
				button = button Or TaskDialogStandardButtons.No
			End If

			If chkClose.Checked Then
				button = button Or TaskDialogStandardButtons.Close
			End If
			If chkRetry.Checked Then
				button = button Or TaskDialogStandardButtons.Retry
			End If

'			#End Region

'			#Region "Icon"

			If rdoError.Checked Then
				td.Icon = TaskDialogStandardIcon.Error
			ElseIf rdoInformation.Checked Then
				td.Icon = TaskDialogStandardIcon.Information
			ElseIf rdoShield.Checked Then
				td.Icon = TaskDialogStandardIcon.Shield
			ElseIf rdoWarning.Checked Then
				td.Icon = TaskDialogStandardIcon.Warning
			End If

'			#End Region

'			#Region "Prompts"

			Dim title As String = txtTitle.Text
			Dim instruction As String = txtInstruction.Text
			Dim content As String = txtContent.Text

'			#End Region

			td.StandardButtons = button
			td.InstructionText = instruction
			td.Caption = title
            td.Text = content
            td.OwnerWindowHandle = Me.Handle

			Dim res As TaskDialogResult = td.Show()

			Me.resultLbl.Text = "Result = " & res.ToString()
		End Sub
	End Class
End Namespace

File: main/contrib/Windows API Code Pack 1.1/source/Samples/TaskDialogDemo/CS/TaskDialogDemo/TestHarness.cs
//Copyright (c) Microsoft Corporation.  All rights reserved.

using System;
using System.Windows.Forms;
using Microsoft.WindowsAPICodePack;
using Microsoft.WindowsAPICodePack.Dialogs;

namespace TaskDialogDemo
{
    public partial class TestHarness : Form
    {
        public TestHarness()
        {
            InitializeComponent();
        }

        private void cmdShow_Click(object sender, EventArgs e)
        {
            TaskDialog td = new TaskDialog();

            #region Button(s)

            TaskDialogStandardButtons button = TaskDialogStandardButtons.None;

            if (chkOK.Checked) button |= TaskDialogStandardButtons.Ok; 
            if (chkCancel.Checked) button |= TaskDialogStandardButtons.Cancel;

            if (chkYes.Checked) button |= TaskDialogStandardButtons.Yes;
            if (chkNo.Checked) button |= TaskDialogStandardButtons.No;

            if (chkClose.Checked) button |= TaskDialogStandardButtons.Close;
            if (chkRetry.Checked) button |= TaskDialogStandardButtons.Retry;

            #endregion

            #region Icon

            if (rdoError.Checked)
            {
                td.Icon = TaskDialogStandardIcon.Error;
            }
            else if (rdoInformation.Checked)
            {
                td.Icon = TaskDialogStandardIcon.Information;
            }
            else if (rdoShield.Checked)
            {
                td.Icon = TaskDialogStandardIcon.Shield;
            }
            else if (rdoWarning.Checked)
            {
                td.Icon = TaskDialogStandardIcon.Warning;
            }

            #endregion

            #region Prompts

            string title = txtTitle.Text;
            string instruction  = txtInstruction.Text;
            string content = txtContent.Text;

            #endregion

            td.StandardButtons = button;
            td.InstructionText = instruction;
            td.Caption = title;
            td.Text = content;
            td.OwnerWindowHandle = this.Handle;
            
            TaskDialogResult res = td.Show();

            this.resultLbl.Text = "Result = " + res.ToString();
        }
    }
}
